ui/src/main/java/org/jtaccuino/core/ui/JavaCellFactory.java

```diff
@@ -471,6 +478,21 @@ private CompletionUpdate convert(int startOfcompletionText, int caretPosition, S
             return new CompletionUpdate(fullCompletionText, remainingCompletion);
         }
 
+        private void handleTabDocumentation(String text, int caretPos, Point2D caretOrigin) {
+            this.control.getSheet().getReactiveJShell().documentationAsync(text, caretPos, result -> {
+                if (result.isEmpty()) {
+                    Platform.runLater(() -> {
+                        documentationPopup.hide();
+                    });
+                } else {
+                    Platform.runLater(() -> {
+                        documentationPopup.setDocumentations(result.stream().map(d -> DocumentationItem.from(d.signature())).toList());
+                        documentationPopup.show(this.control.getScene().focusOwnerProperty().get(), caretOrigin);
+                    });
+                }
```

Comment:
You could move the if-else-statement into the lambda run on the Platform.
e.g.:

```suggestion
                Platform.runLater(() -> {
                    if (result.isEmpty()) {
                        documentationPopup.hide();
                    } else {
                        documentationPopup.setDocumentations(result.stream().map(d -> DocumentationItem.from(d.signature())).toList());
                        documentationPopup.show(this.control.getScene().focusOwnerProperty().get(), caretOrigin);
                    });
                }
```